contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -4860,4 +4860,78 @@ LANGUAGE plpgsql STABLE;
 CREATE OR REPLACE FUNCTION sys.bbf_pivot()
 RETURNS setof record
 AS 'babelfishpg_tsql', 'bbf_pivot'
-LANGUAGE C STABLE;
\ No newline at end of file
+LANGUAGE C STABLE;
+
+-- wrapper functions for reverse
+CREATE OR REPLACE FUNCTION sys.reverse(string ANYELEMENT)
+RETURNS sys.VARCHAR
+AS
+$BODY$
+DECLARE
+    string_arg_datatype text;
+    string_basetype oid;
+BEGIN
+    string_arg_datatype := sys.translate_pg_type_to_tsql(pg_typeof(string)::oid);
```

Comment:
cache the value of pg_typeof(string)::oid

Reply:
Done